**Задача 1**

Из партии батареек с номинальным напряжением 12 В наугад выбрано семь штук. Согласно показаниям высокоточного вольтметра, фактически выдаваемое ими напряжение составило: 12,9; 11,6; 13,5; 13,9; 12,1; 11,9; 13,0 В соответственно. Можно ли считать, что среднее напряжение действительно равно 12 В?

In [1]:
import numpy as np
from scipy import stats

Гипотезы:

H0: μ = 12

H1: μ ≠ 12

In [2]:
sample = np.array([12.9, 11.6, 13.5, 13.9, 12.1, 11.9, 13])

Проверим данные на нормальность с помощью критерия Шапиро-Уилка

In [3]:
stats.shapiro(sample)

ShapiroResult(statistic=0.9461528332649155, pvalue=0.694601251101445)

H0 о нормальности не отвергается

Используем Одновыборочный критерий Стьюдента

In [4]:
mu0 = 12

In [5]:
df = len(sample)-1
student = stats.t(df)
T = (df)**0.5*(sample.mean() - mu0)/sample.std()
P = (1 - student.cdf(T))*2
print("Statistic: ",T,", p-value: ",P)

Statistic:  2.1578017974978794 , p-value:  0.07430190210562304


p-value = 0.074 > 0.05. H0 не отвергается. Значит, мы можем считать, что среднее напряжение равно 12.

**Задача 2**

Есть два различных метода определения процентного содержания жира в мясе. Доля жира для 10 образцов мяса была определена каждым из двух методов. Позволяют ли полученные результаты сделать вывод о том, что один метод даёт в среднем более высокие показания, чем другой?

Образец

1    2    3    4    5    6    7   8    9    10

Метод I

23,1 23,2 26,5 26,6 27,1 48,3 40,5 25,0 38,4 23,5

Метод II

22,7 23,6 27,1 27,4 27,4 46,8 40,4 24,9 38,1 23,8

Гипотезы:

H0: μ1 = μ2 (методы дают одинаковые показатели)

H1: μ1 > μ2 (первый метод дает более высокие показатели, чем 2)

In [6]:
sample1 = np.array([23.1, 23.2, 26.5, 26.6, 27.1, 48.3, 40.5, 25, 38.4, 23.5])
sample2 = np.array([22.7, 23.6, 27.1, 27.4, 27.4, 46.8, 40.4, 24.9, 38.1, 23.8])

Проверим данные на нормальность с помощью критерия Шапиро-Уилка. Так как выборка по сути у нас 1, а не 2 независимых, то мы проверяем нормальность (sample1-sample2) значений.

In [7]:
stats.shapiro(sample1-sample2)

ShapiroResult(statistic=0.8998483579792755, pvalue=0.21825780478589868)

H0 о нормальности не отвергается

Используем Двухвыборочный критерий Стьюдента для парных выборок

In [8]:
d = sample1 - sample2
df = len(d)-1
t = stats.t(df)
T = (df)**0.5*(d.mean())/d.std()
P = 1 - t.cdf(T)
print("Statistic: ",T,", p-value: ",P)

Statistic:  1.715488497867803e-15 , p-value:  0.49999999999999933


p-value = 0.499(9) > 0.05. H0 не отвергается. Мы не можем сказаь, что 1 метод дает более высокие показатели.

**Задача 3**

Для проверки гипотезы о том, что жители городов имеют в среднем более высокое систолическое артериальное давление, чем жители сельской местности, было проведено обследование 10 горожан и 9 жителей села примерно одного возраста. Получены следующие данные об уровне давления для горожан: 132, 111, 119, 138, 200, 131, 138, 170, 159, 140; для селян: 115, 190, 127, 155, 148, 121, 116, 121, 197. Можно ли считать верной выдвинутую гипотезу на уровне значимости 0,05?

Гипотезы:

H0: μ1 = μ2

H1: μ1 > μ2

In [9]:
x = np.array([132, 111, 119, 138, 200, 131, 138, 170, 159, 140])
y = np.array([115, 190, 127, 155, 148, 121, 116, 121, 197])

Проверим данные на нормальность с помощью критерия Шапиро-Уилка


In [10]:
stats.shapiro(x)

ShapiroResult(statistic=0.9100654196723891, pvalue=0.281440686937816)

In [11]:
stats.shapiro(y)

ShapiroResult(statistic=0.8286640993349573, pvalue=0.04313951145476069)

данные во 2ой выборке не нормальны (вообще строго проверка каждого из 2ух распределений на нормальность не равно нормальности совместного распределения, но мы это опустим). так как 2ое распределение не нормально, будем использовать критерий Манна-Уитни

In [12]:
m = len(x)
n = len(y)
xy = np.hstack([x, y])
r = stats.rankdata(xy)
print(xy)
print(r)
ww = r[:m]
W = sum(ww)
ew = m*(m + n + 1)/2
varw = m*n*(m + n + 1)/12
WT = (W - ew)/varw**0.5
P = 1 - stats.norm.cdf(WT)
print("Statistic: ",WT,", p-value: ",P)

[132 111 119 138 200 131 138 170 159 140 115 190 127 155 148 121 116 121
 197]
[ 9.   1.   4.  10.5 19.   8.  10.5 16.  15.  12.   2.  17.   7.  14.
 13.   5.5  3.   5.5 18. ]
Statistic:  0.408248290463863 , p-value:  0.34154569915480437


p-value = 0.3415 > 0.05. H0 не отвергается. Значит, мы не можем считать, что жители городов имеют в среднем более высокое систолическое артериальное давление, чем жители сельской местности (так как нулевая гипотеза у нас о равенстве средних).

**Задача 4**

Известно, что 52% новорождённых детей — мальчики, а 48% — девочки. В случайной выборке из 5 000 человек в возрасте от 30 до 60 лет мужчин оказалось 2 500. Можно ли на уровне значимости 0,05 считать, основываясь на этих данных, что смертность среди мужчин и женщин одинакова?

Гипотезы:

H0: p1 = p2 (нет различий в смертности между мужчинами и женщинами, т.е. доли равны)

H1: p1 ≠ p2 (различия в смертности есть)

применим Биномиальный критерий для проверки гипотезы о равенстве долей/вероятностей

In [13]:
n1 = 2500
m1 = 2500 * 0.52
n2 = 2500
m2 = 2500 * 0.48

In [14]:
T = (m1/n1 - m2/n2)/((m1+m2)/(n1+n2)*(1 - (m1+m2)/(n1+n2))*(1/n1 + 1/n2))**0.5
P = (1 - stats.norm.cdf(T))*2
print("Statistic: ",T,", p-value: ",P)

Statistic:  2.8284271247461925 , p-value:  0.004677734981047177


p-value = 0.0047 < 0.05. H0 отвергается. Мы не можем считать, что смертность среди мужчин и женщин одинакова.

**Задача 5**

В первом эксперименте для сравнения двух средних X1=17, X2=16 применялся t-критерий Стьюдента, и эти различия оказались значимы (p = 0, 001).

Во втором исследовании, также при помощи t - критерия, сравнивались два средних X1=17, X2=36, и эти различия не значимы при p = 0,8. В чем может быть причина таких результатов, при условии, что нулевая и альтернативная гипотезы в обоих случаях одинаковы?

Причина таких результатов может быть в следующем:


1. Дисперсия в данных.

Если во втором эксперименте данные имеют большую вариативность (дисперсию) по сравнению с первым исследованием, то это может привести к тому, что даже большая разница в средних (17 и 36) не будет статистически значимой, поскольку t-критерий чувствителен к вариативности внутри групп.



2. Размер выборки.

Если в первом эксперименте размер выборки был значительно меньше, а во втором — значительно больше, то это могло повлиять на результаты. При большем размере выборки статистическая мощность теста (способность обнаружить эффект, если он существует) будет выше. Поэтому если дисперсия была высокой во втором эксперименте, то даже значительная разница между средними (17 и 36) могла стать статистически незначимой.